src/PanelTraits/Update.php

```diff
@@ -73,4 +74,22 @@ public function getUpdateFields($id)
 
         return $fields;
     }
+
+    private function getEntryValue($entry, $field)
+    {
+        if (isset($field['entity'])) {
+            $relationArray = explode('.', $field['entity']);
+            $relatedModel = array_reduce(array_splice($relationArray, 0, -1), function ($obj, $method) {
+                return $obj->{$method} ? $obj->{$method} : $obj;
+            }, $entry);
+
+            if ($relatedModel->{end($relationArray)} && $relatedModel->{end($relationArray)}() instanceof HasOneOrMany) {
```

Comment:
no.. i still don't see why we need both the checks.. the second one is enough. Also shifting it to a vaiable would be much better and avoid calling the same function `end` twice